lib/browser/api/notification.js

```diff
@@ -1,8 +1,72 @@
+const {app} = require('electron')
 const {EventEmitter} = require('events')
-const {Notification, isSupported} = process.atomBinding('notification')
+const {Notification: NativeNotification, isSupported} = process.atomBinding('notification')
+const querystring = require('querystring')
+const path = require('path')
+const url = require('url')
 
-Object.setPrototypeOf(Notification.prototype, EventEmitter.prototype)
+Object.setPrototypeOf(NativeNotification.prototype, EventEmitter.prototype)
+
+let providedProtocol
+let notificationId = 1
+const notificationMap = {}
```

Comment:
This map is going to grow infinitely, it should be cleared when an notification is destroyed. You can reference how `atom_api_download_item.cc` did bookkeeping.